dsgrid/dataset/dimension_filters.py

```diff
@@ -0,0 +1,188 @@
+import abc
+import logging
+from typing import Any, Union
+
+import pyspark.sql.functions as F
+from pydantic import Field, validator
+
+from dsgrid.data_models import DSGBaseModel
+from dsgrid.dimension.base_models import DimensionType
+
+
+logger = logging.getLogger(__name__)
+
+
+class DimensionFilterBaseModel(DSGBaseModel, abc.ABC):
+
+    dimension_type: DimensionType
+    query_name: str
+
+    @abc.abstractmethod
+    def apply_filter(self, df):
+        """Apply the filter to a DataFrame"""
+
+    @staticmethod
+    def _make_value_str(value):
+        if isinstance(value, str):
+            return f"'{value}'"
+        return str(value)
+
+    @staticmethod
+    def _make_values_str(values):
+        if isinstance(values[0], str):
+            text = ", ".join((f"'{x}'" for x in values))
+        elif isinstance(values[0], int) or isinstance(values[0], float):
+            text = ", ".join((f"{x}" for x in values))
+        else:
+            raise Exception(f"Unsupported type: {type(values[0])}")  # TODO: dsg exception
+
+        return f"({text})"
+
+
+class _DimensionFilterWithWhereClauseModel(DimensionFilterBaseModel, abc.ABC):
+    def apply_filter(self, df, column=None):
+        return df.filter(self.where_clause(column=column))
+
+    @abc.abstractmethod
+    def where_clause(self, column=None):
+        """Returns the text for a where clause in a filter statement.
+
+        Parameters
+        ----------
+        column : None or str
+            Column to use. If None, use the dimension type.
+
+        Returns
+        -------
+        str
+
+        """
+
+
+class DimensionFilterValueModel(_DimensionFilterWithWhereClauseModel):
+    """Filters a table where a dimension column has a specific value.
+
+    Example:
+        DimensionFilterValueModel(
+            dimension_type=DimensionType.GEOGRAPHY,
+            query_name="county",
+            value="06037",
+        ),
+    is equivalent to
+        df.filter("county == '06037'")
+
+    """
+
+    value: Any
+
+    def where_clause(self, column=None):
+        if column is None:
+            column = self.dimension_type.value
+        value = self._make_value_str(self.value)
+        text = f"({column} == {value})"
+        return text
+
+
+class DimensionFilterExpressionModel(_DimensionFilterWithWhereClauseModel):
+    """Filters a table where a dimension column matches an expression.
+    Builds the filter string based on inferred types.
+
+    Example:
+        DimensionFilterExpressionModel(
+            dimension_type=DimensionType.GEOGRAPHY,
+            query_name="county",
+            operator="=="
+            value="06037",
+        ),
+    is equivalent to
+        df.filter("county == '06037'")
+
```

Comment:
This example is the same example that you have in `DimensionFilterValueModel`. I am a little confused in the difference between the two, so adding another example (or modifying this one) to distinguish the two classes would be helpful. 

... Also it looks like Expression has the `operator` while Value does not. Is that because Value is an implied `==` operator?

Reply:
DimensionFilterValueModel is a shortcut for `==`. I can remove it if it's confusing. You can use any operator in DimensionFilterExpressionModel.

Reply:
Remove DimensionFilterValueModel